### ISSUE TITLE
Support custom metrics in progress bar format

### ISSUE THREAD

**Issue by ferrouswheel** [tqdm/tqdm#266]
It'd be nice to update custom metrics on the progress bar status. This would make tqdm super helpful for something like machine learning training where you want to know how well your model is doing.

e,g. before:

```
0%|3                                             | 27/6000 [08:24<28:42:16, 17.30s/it]
```

after:

```
0%|3                                | 27/6000 [08:24<28:42:16, 17.30s/it, 0.1233 loss]
```

**Comment by lrq3000** [comment]
Hello @ferrouswheel,

You can supply additional information in the description:

```
from tqdm import trange
t = trange(6000, desc='ML')
for i in trange:
    loss = compute_loss() # your function here
    t.set_description('ML (loss=%g)' % loss)
```

We could add a `set_postfix()` function to allow you to add information at the end of the bar, but this would be a slight additional overhead.

**Comment by lrq3000** [comment]
PS: please look the previous comment on github, not in the email, I fixed an error in the code.

**Comment by CrazyPython** [comment]
@lrq3000 this would be at the left though - maybe add a `set_postfix()` function in a submodule? Or a simple subclass recipe.

**Comment by ferrouswheel** [comment]
@lrq3000 Thanks - that's awesome :+1: 

I think I assumed it should go on the right because unless the loss text width is fixed it'll make the progress bar jump around a bit, but that's easy enough to avoid.

The implementation I thought might be most powerful is being able to provide tqdm a template "context", which the progress bar format string could reference variables from. Not sure what the performance implications are though....

**Comment by lrq3000** [comment]
@ferrouswheel Please see if #270 meets your needs better than `set_description()`.

**Comment by ferrouswheel** [comment]
Thanks for implementing this!
